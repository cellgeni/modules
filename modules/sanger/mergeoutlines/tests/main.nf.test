nextflow_process {

    name "Test Process MERGEOUTLINES"
    script "../main.nf"
    config "./nextflow.config"
    process "MERGEOUTLINES"

    tag "modules"
    tag "modules_sanger"
    tag "mergeoutlines"

    test("mouse heart - wkt") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    [
                        file(params.modules_testdata_base_path + 'test-0_0_200_200-diam_20_cp_outlines.wkt', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'test-0_0_500_500-diam_20_cp_outlines.wkt', checkIfExists: true),
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse heart - wkt - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    [
                        file(params.modules_testdata_base_path + 'test-0_0_200_200-diam_20_cp_outlines.wkt', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'test-0_0_500_500-diam_20_cp_outlines.wkt', checkIfExists: true),
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
