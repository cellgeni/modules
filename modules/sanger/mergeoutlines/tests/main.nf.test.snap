{
    "mouse heart - wkt - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,b47d0d0a777ffdcc4fc0c230c150c040"
                ],
                "multipoly_geojsons": [
                    
                ],
                "multipoly_wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b47d0d0a777ffdcc4fc0c230c150c040"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-15T10:04:50.507810178"
    },
    "mouse heart - wkt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,ac7ba24ae1deff214bf465aed36a42f9"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,0e17590b6af5065d58270ee8708db50c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,b47d0d0a777ffdcc4fc0c230c150c040"
                ],
                "multipoly_geojsons": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,0e17590b6af5065d58270ee8708db50c"
                    ]
                ],
                "multipoly_wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,ac7ba24ae1deff214bf465aed36a42f9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b47d0d0a777ffdcc4fc0c230c150c040"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-15T10:04:43.098270886"
    }
}